File: Implemetation, Math/C++/1748 수 이어 쓰기 1.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int n;
    cin >> n;
    int d = int(log10(float(n))+1);
    long long answer = 0;
    int temp = n;
    int firstDigit = 0;
    int elseDigit = 0;
    while(temp >= 10) {
        temp /= 10;
    }
    firstDigit = temp;
    elseDigit = n % int(pow(10,d-1));
    int value = 0;
    for(int i = 0 ; i < d; i++) {
        if(i==0) continue;
        else {
            value += (i * int(pow(10, i-1))) * 9;
        }
    }
    
    answer = value + ((d*int(pow(10,d-1))) * (firstDigit-1)) + ((elseDigit+1) * d);
    
    cout << answer;
}

File: Implemetation, Math/Kotlin/1748 수 이어 쓰기 1.kt
import java.util.*;
import kotlin.math.*;

fun findDigit(number : Int) : Int {
    var result = 0; var n = number
    while(n > 0) {
        n /= 10
        result += 1
    }
    return result
}

fun main(args: Array<String>) {
    var sc = Scanner(System.`in`)
    var n = sc.nextInt(); var ten = 10f;
    var answer = 0; var d = findDigit(n); var temp = n;
    var v = 0;
    while(temp >= 10) {temp/= 10};
    var f = temp; var e  = n % (ten.pow(d-1).toInt())
    for(i in 1 until d) {v += (i * (ten.pow(i-1)).toInt()) * 9}
    answer = v + ((d* (ten.pow(d-1)).toInt())) * (f-1) + (e+1)*d

    print(answer)
}
